src/content/docs/apm/agents/java-agent/api-guides/ai-monitoring.mdx

```diff
@@ -0,0 +1,99 @@
+
+    * `trace_id` (required) - ID of the trace where the chat completion(s) related to the feedback occurred
+    * `rating` (required) - Rating provided by an end user (ex: 'Good', 'Bad', 1, 2, 5, 8, 10)
+    * `category` (optional) - Category of the feedback as provided by the end user (ex: “informative”, “inaccurate”)
+    * `message` (optional) - Freeform text feedback from an end user
+    * `metadata` (optional) - Set of key-value pairs to store any other desired data to submit with the feedback event
+
+If the user feedback is to be recorded on a different thread, or even in a completely different service, from where the LLM prompt/response occurred, then the trace ID will need to be acquired from the originating thread or service and propagated to where the user feedback event will be recorded. The [`TraceMetadata.getTraceId()`](https://newrelic.github.io/java-agent-api/javadoc/com/newrelic/api/agent/TraceMetadata.html#getTraceId) API can be used to acquire the trace ID for the currently executing transaction.
+```java
+String traceId = NewRelic.getAgent().getTraceMetadata().getTraceId();
+```
+
+Below is a complete example of how you could record an LLM feedback event:
+```java
+String traceId = ... // acquired directly from New Relic API or retrieved from some propagation mechanism
+
+Map<String, String> metadata = new HashMap<>();
+metadata.put("interestingKey", "interestingVal");
+
+LlmFeedbackEventAttributes.Builder llmFeedbackEvenAttrBuilder = new LlmFeedbackEventAttributes.Builder(traceId, ratingString);
+
+Map<String, Object> llmFeedbackEventAttributes = llmFeedbackEvenAttrBuilder
+        .category("General")
+        .message("Ok")
+        .metadata(metadata)
+        .build();
+
+NewRelic.getAgent().getAiMonitoring().recordLlmFeedbackEvent(llmFeedbackEventAttributes);
+```
+
+## Adding custom LLM attributes [#custom-llm-attributes]
+
+Any custom attributes added using the [`NewRelic.addCustomParameter(...)`](https://newrelic.github.io/java-agent-api/javadoc/com/newrelic/api/agent/NewRelic.html#addCustomParameter(java.lang.String,boolean)) API that are prefixed with `llm.` will automatically be copied to `LlmEvent`s. For custom attributes added by the `addCustomParameters` API to be added to `LlmEvent`s the API calls must occur before invoking the Bedrock SDK.
+
+One optional custom attribute with special meaning is `llm.conversation_id`, which can be used to group LLM messages into specific conversations in APM.
+
+## LLM event limits [#llm-event-limits]
```

Comment:
i'm removing this from this doc since these events + attributes are in the data dictionary, and they aren't strictly related to the API itself. The limits should appear in the Java configuration doc instead